aten/src/ATen/core/jit_type.h

```diff
@@ -1325,25 +1325,31 @@ inline TypePtr CompleteTensorType::fromNumberType(TypePtr typ) {
   } else if (typ->isSubtypeOf(BoolType::get())) {
     return CompleteTensorType::create(at::kLong, at::kCPU, {});
   }
-  AT_ERROR("unknown number type", typ->str());
+  AT_ERROR("unknown number type: ", typ->str());
 }
 
 inline TypePtr CompleteTensorType::fromBoolType() {
   return CompleteTensorType::create(at::kLong, at::kCPU, {});
 }
 
-inline at::ScalarType scalarTypeFromJitType(const c10::TypePtr& type) {
+inline c10::optional<c10::ScalarType> optionalScalarTypeFromJitType(const c10::TypePtr & type) {
```

Comment:
Changes here look good. Nit: in our codebase we usually have `foo` and `tryFoo` for an erroring / optional version of the same function. E.g. `tryInsertConstant` & `insertConstant`